### PR TITLE
Pivotal ID # 178028214: Show requests in submissions list endpoint

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/178028214

### CODE REVIEW

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/model/DocSubmissionExt.kt [hunk 1]
```diff
@@ -34,3 +39,35 @@ private fun DocSubmissionMethod.toSubmissionMethod(): SubmissionMethod =
         DocSubmissionMethod.PAGE_TAB -> SubmissionMethod.PAGE_TAB
         DocSubmissionMethod.UNKNOWN -> SubmissionMethod.UNKNOWN
     }
+
+fun ExtSubmission.asBasicSubmission(): BasicSubmission {
```
Comment (jhoanmanuelms) [inline submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/model/DocSubmissionExt.kt:43]: one-liner

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/model/DocSubmissionExt.kt [hunk 2]
```diff
@@ -34,3 +39,35 @@ private fun DocSubmissionMethod.toSubmissionMethod(): SubmissionMethod =
...
+val ExtSection.title: String?
+    get() = attributes.find { it.name == SectionFields.TITLE.value }?.value
+
+private fun ExtProcessingStatus.toProcessingStatus(): ProcessingStatus =
+    when (this) {
+        ExtProcessingStatus.PROCESSED -> ProcessingStatus.PROCESSED
+        ExtProcessingStatus.PROCESSING -> ProcessingStatus.PROCESSING
+        ExtProcessingStatus.REQUESTED -> ProcessingStatus.REQUESTED
+    }
+private fun ExtSubmissionMethod.toSubmissionMethod(): SubmissionMethod =
```
Comment (jhoanmanuelms) [inline submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/model/DocSubmissionExt.kt:68]: new line

File: submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoQueryServiceTest.kt [hunk 3]
```diff
@@ -24,18 +36,19 @@ import org.testcontainers.junit.jupiter.Container
...
 @SpringBootTest(classes = [MongoDbReposConfig::class])
 internal class SubmissionMongoQueryServiceTest(
     @MockK private val toExtSubmissionMapper: ToExtSubmissionMapper,
-    @MockK private val serializationService: ExtSerializationService,
     @Autowired private val submissionRepo: SubmissionDocDataRepository,
     @Autowired private val requestRepository: SubmissionRequestDocDataRepository
 ) {
     private val testInstance =
-        SubmissionMongoQueryService(submissionRepo, requestRepository, serializationService, toExtSubmissionMapper)
+        SubmissionMongoQueryService(submissionRepo, requestRepository, ExtSerializationService(), toExtSubmissionMapper)
```
Comment (jhoanmanuelms) [inline submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoQueryServiceTest.kt:51]: As far as I can see, the serialized submission isn't actually being used nor asserted so I think it should be mocked.

File: submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoQueryServiceTest.kt [hunk 4]
```diff
@@ -45,6 +58,159 @@ internal class SubmissionMongoQueryServiceTest(
...
+
+        private fun saveAsRequest(extSubmission: ExtSubmission): ExtSubmission {
+            requestRepository.saveRequest(asRequest(extSubmission))
+            return extSubmission
+        }
+
+        private fun asRequest(submission: ExtSubmission) = SubmissionRequest(
+            accNo = submission.accNo,
+            version = submission.version,
+            submission = BasicDBObject.parse(ExtSerializationService().serialize(submission))
```
Comment (jhoanmanuelms) [inline submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoQueryServiceTest.kt:210]: There's no neeed to create a new serialization service instance

**Review by jhoanmanuelms — COMMENTED** [review]
I've got a couple of comments

File: submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoQueryServiceTest.kt (hunk 4 above)
Comment (Juan-EBI) [reply submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoQueryServiceTest.kt:210]: great catch

**Review by jhoanmanuelms — APPROVED** [review]
LGTM

File: submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoQueryServiceTest.kt (hunk 3 above)
Comment (Juan-EBI) [reply submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoQueryServiceTest.kt]: it is used now to setup data

**Pull request merged**